### DISCUSSION TITLE
Not authorized to perform sts:AssumeRoleWithSAML

### DISCUSSION THREAD

**Discussion by SyndrixLane** [amazon-connect/amazon-connect-salesforce-scv#137] [Q&A] [answered]
Hello,

I am now setting everything up in our production environment,
I had salesforce build a new aws connect instance, but when I try to log in through salesforce using the IDP initiated url/telephony provider settings I am getting this error
Not authorized to perform sts:AssumeRoleWithSAML (Service: AWSSecurityTokenV20111201; Status Code: 403; Error Code: AccessDenied; Request ID: addb758f-6a3e-421f-b157-7de43b72cac8; Proxy: null). Please try again.

I have checked the saml role that was created when the contact center was created 
The trust policy looks correct
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "Federated": "removed for security/arn points to the idp"
            },
            "Action": "sts:AssumeRoleWithSAML",
            "Condition": {
                "StringEquals": {
                    "saml:aud": "https://signin.aws.amazon.com/saml"
                }
            }
        }
    ]
}
What else could I check/change?
As always, thank you for your help and time.

**Comment by SyndrixLane** [comment, marked as answer]
I was able to correct this issue, the federated arn was pointing to the wrong identity provider arn in aws for some reason.
Feel free to delete/remove this thread.

**Comment by enilvia** [comment]
Thanks for this- helped after spending a lot of time troubleshooting.
